datahub-web-react/src/app/entity/shared/siblingUtils.ts

```diff
@@ -51,9 +51,16 @@ const combineMerge = (target, source, options) => {
     return destination;
 };
 
-// use when you want to merge and array of objects by key in the object as opposed to by index of array
+function convertObjectKeysToLowercase(object: Record<string, unknown>) {
+    return Object.fromEntries(Object.entries(object).map(([key, value]) => [key.toLowerCase(), value]));
```

Comment:
It's a bit confusing - what is the structure of this?

We take a recored object, take its keys, then return the key lower cased but not the value... Why does this help with a list of objects like

[
{ fieldPath: "lower" ...}
{ fieldPath: "upper" ...}
]

or are these keys and values different from what im imagining?

Reply:
yup so the shape of the object we get is described below where the `fieldPath` value is going to be the key, and the object is going to be the value.

What this method is doing is it simply converts all of those keys in the object described below to lowercase, otherwise it leaves it all the same.

Building off the example below, if the original incoming array looked like this:
```
[{ fieldPath: 'USER_NAME', ... }, { fieldPath: 'user_id', ... }, { fieldPath: 'user_email', ... }]
```
and then we get this object:
```
{
    USER_NAME: { fieldPath: 'USER_NAME', ... },
    user_id: { fieldPath: 'user_id', ... },
    user_email: { fieldPath: 'user_email', ... }
}
```
then this method will return this:
```
{
    user_name: { fieldPath: 'USER_NAME', ... },
    user_id: { fieldPath: 'user_id', ... },
    user_email: { fieldPath: 'user_email', ... }
}
```

Reply:
nice - thanks for the thorough explanation!